File: spec/javascript/test_nested_same_name.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('NestedSameName', 'src/repeat_n_struct.bin', function(r) {
  assert.equal(r.mainData.mainSize, 2);
  assert.equal(r.mainData.foo.data.toString(), [0x10, 0, 0, 0].toString());
});

File: spec/python/test_nested_same_name.py
import unittest

from nested_same_name import NestedSameName

class TestNestedSameName(unittest.TestCase):
    def test_nested_same_name(self):
        with NestedSameName.from_file("src/repeat_n_struct.bin") as r:

            self.assertEqual(r.main_data.main_size, 2)
            self.assertEqual(r.main_data.foo.data, b"\x10\0\0\0")
